### PR TITLE
Suppress duplicate AI agents

### LINKED ISSUE #1345: Duplicated log events in AppInsights dashboard
We tried using AppInsights agent 3.0.0-PREVIEW.7 to push log4j events for a JavaApplication server to see INFO level logs reported, the events are logged as expected but we are now seeing duplicated entries in the AppInsights traces. 

### Expected behavior
Server logs appear in the AppInsights dashboard at all levels

### Actual behavior
Server logs appear but with a duplicate written for each

### To Reproduce
1. Start the java application server using Log4j2 with jvm arg `-javaagent applicationinsights-agent-3.0.0-PREVIEW.7.jar`
2. Observe that events can be seen duplicated in ApplicationInsights traces

## Log4j configuration
We thought this issue might have something to do with having multiple loggers receiving events with `additivity="false"` but it's been reproducible with a fully stripped back configuration comprising a single Logger and Appender.
```
<Configuration status="error" name="foo" monitorInterval="300">
  
  <Appenders>
    <RollingFile name="foo" fileName="${env:LOG_DIR:-${sys:app.home:-/tmp}/logs}/foo-server.log"
                 filePattern="$${env:LOG_DIR:-$${sys:app.home:-/tmp}/logs}/foo-server-%d{yyyy-MM-dd'T'HH:mm:ss}{UTC}.log">
      <PatternLayout pattern="%d{ISO8601}{UTC}|%-5p|%t|%c{1.}|$${ctx:CORRELATION_ID:-N/A}|%m%n" charset="UTF-8"/>
      <Policies>
        <OnStartupTriggeringPolicy minSize="1"/>
        <TimeBasedTriggeringPolicy interval="86400" modulate="true"/>
        <SizeBasedTriggeringPolicy size="200 MB"/>
      </Policies>
      <DefaultRolloverStrategy>
        <Delete basePath="${env:LOG_DIR:-${sys:app.home:-/tmp}/logs}">
          <IfFileName glob="foo-server-*.log"/>
          <IfAccumulatedFileCount exceeds="26"/>
        </Delete>
      </DefaultRolloverStrategy>
    </RollingFile>
  </Appenders>
  
  <Loggers>
    <Root level="${env:ROOT_LOG_LEVEL:-INFO}">
      <AppenderRef ref="foo"/>
    </Root>
  </Loggers>
  
</Configuration>
```

### System information
Please provide the following information:
 - SDK Version: 3.0.0-PREVIEW.7
 - OS type and version: MacOS Mojave 10.14.6
 - Application Server type and version (if applicable): Java (Jetty 9.4.19.v20190610)
 - Using spring-boot? No
 - Additional relevant libraries (with version, if applicable): Log4j 2.12.1

### Screenshots
Screenshot 2020-10-09 at 17 23 29

Unsure at what point it is events are being duplicated, ingest or egest. I'm monitoring this issue and can follow through on producing any useful debugging on the server side. We had been pushing our logging successfully using `com.microsoft.azure:applicationinsights-logging-log4j2:2.6.1` prior.

### PR DESCRIPTION
**Pull request by trask**
This PR makes sure that, as long as the 3.0.0 java agent is the first `-javaagent` on the command line, that it will suppress other Application Insights java agents.

This is primarily helpful for App Services attach, where we use `JAVA_TOOL_OPTIONS` (which is prepended to the command line args).

This also resolves #1345 😄 (https://stackoverflow.com/questions/60838748/intellij-idea-java-instrumentation-premain-gets-called-twice-when-directly-run).

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/DuplicateAgentClassFileTransformer.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+public class DuplicateAgentClassFileTransformer implements ClassFileTransformer {
+
+    private static final Logger logger = LoggerFactory.getLogger(DuplicateAgentClassFileTransformer.class);
+
+    // using constant here so that it will NOT get shaded
+    // IMPORTANT FOR THIS NOT TO BE FINAL (or private)
+    // OTHERWISE COMPILER COULD THEORETICALLY INLINE IT BELOW AND APPLY .substring(1)
+    // and then it WOULD be shaded
+    static String[] UNSHADED_CLASS_NAMES = new String[] {
+            "!io/opentelemetry/javaagent/OpenTelemetryAgent", // 3.0
```
Comment (kryalama) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/DuplicateAgentClassFileTransformer.java:48]: Do we need this line

**Review by heyams — COMMENTED** [review]
is it possible to add a test for detecting duplicate agents?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/DuplicateAgentClassFileTransformer.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/DuplicateAgentClassFileTransformer.java]: nope, removed, and added comment to OpenTelemetryAgent

**Comment by trask** [comment]
> is it possible to add a test for detecting duplicate agents?

ya, you're right, this is important, I added a backlog item to add test: https://msazure.visualstudio.com/One/_workitems/edit/9112258

**Review by heyams — APPROVED** [review]
